### PR TITLE
Yet another haskey method

### PR DESCRIPTION
**Pull request by heetbeet**
The mother of all VBA hasKey methods

### CODE REVIEW

**Comment by malankriel** [comment]
@heetbeet we don't expect all the extra checking to have significant runtime impacts here?

**Comment by malankriel** [comment]
If any, will surely be negligible :)

File: MiscVBAFunctions/Modules/MiscHasKey.bas [hunk 1]
```diff
@@ -8,43 +8,110 @@ Private Sub TestHasKey()
     c.Add col("x", "y", "z"), "b"
     
     Debug.Print vbLf & "*********** TestHasKey tests ***********"
-    Debug.Print hasKey(c, "a") ' True for scalar
-    Debug.Print hasKey(c, "b") ' True for object
-    Debug.Print hasKey(c, "A") ' False (case insensitive)
+    Debug.Print True, hasKey(c, "a") ' True for scalar
+    Debug.Print True, hasKey(c, "b") ' True for object
+    Debug.Print True, hasKey(c, "A") ' True (even though case insensitive???)
```
Comment (malankriel) [inline MiscVBAFunctions/Modules/MiscHasKey.bas:13]: Yes, I had it wrong, it will return `True` for collections as it's case insensitive, right?

**Comment by heetbeet** [comment]
I don't think it will have an impact. I see that collection has a `.Item` method, so the bottom `()` check will rarely be activated. Dictionaries will short-circuit at the `Exists` check and be the fastest check.

**Comment by heetbeet** [comment]
Minor impact at least. There is lost of code but not that much work really going on, except for the errors. I don't know how efficient errors are in VBA

File: MiscVBAFunctions/Modules/MiscHasKey.bas (hunk 1 above)
Comment (heetbeet) [reply MiscVBAFunctions/Modules/MiscHasKey.bas:13]: Yes it returns True on my checks

**Pull request merged**